src/tools/perf/perftest_params.c

```diff
@@ -253,6 +267,95 @@ static ucs_status_t parse_ucp_datatype_params(const char *opt_arg,
     return UCS_OK;
 }
 
+static ucs_status_t set_sockaddr(const char *ip_addr_str, uint16_t port,
+                                 struct sockaddr_storage *addr)
+{
+    struct sockaddr_in *sa_in   = (struct sockaddr_in*)addr;
+    struct sockaddr_in6 *sa_in6 = (struct sockaddr_in6*)addr;
+
+    if (inet_pton(AF_INET, ip_addr_str, &sa_in->sin_addr) == 1) {
+        sa_in->sin_family = AF_INET;
+        sa_in->sin_port   = htons(port);
+        return UCS_OK;
+    }
+
+    if (inet_pton(AF_INET6, ip_addr_str, &sa_in6->sin6_addr) == 1) {
+        sa_in6->sin6_family = AF_INET6;
+        sa_in6->sin6_port   = htons(port);
+        return UCS_OK;
+    }
+
+    ucs_error("invalid address '%s'", ip_addr_str);
+
+    return UCS_ERR_INVALID_PARAM;
+}
+
+static ucs_status_t parse_ip_addr_port(const char *str,
+                                       struct sockaddr_storage *addr,
+                                       uint16_t port)
```

Comment:
Well, the idea was to make a reusable function, so that we can use it in other places.
This pattern `IP:port` seems to be quite common.
What do you think about moving this function to `ucs/sock` module?

Reply:
Btw, for IPv6 addresses this string representation `IP:port` is ambiguous.
It seems we will have to use this format for IPv6: `[IPv6]:port`, which as [SO say](https://softwareengineering.stackexchange.com/questions/311413/how-should-we-represent-an-ipv6-address-with-port-number-in-text) is used in RFCs 5952, 3986